streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueStoreSupplier.java

```diff
@@ -117,6 +117,15 @@ public void put(K key, V value) {
         }
 
         @Override
+        public V putIfAbsent(K key, V value) {
+            V originalValue = get(key);
```

Comment:
I see this can be simpler if `put` call itself can also return the old value. But this would better be in a separate PR.